src/delagent/agent/util.c

```diff
@@ -28,148 +29,324 @@ int Verbose = 0;
+  *user_id = atoi(PQgetvalue(result, 0, 3));
+  PQclear(result);
+  if (user_seed[0] && pass_hash_valid[0])
+  {
+    strcat(user_seed, password);  // get the hash code on seed+pass
+    SHA1((unsigned char *)user_seed, strlen(user_seed), pass_hash_actual_raw);
+  }
+  else
+  {
+    return -1;
+  }
+  int i = 0;
+  char temp[256] = {0};
+  for (i = 0; i < 20; i++)
+  {
+    snprintf(temp, 256, "%02x", pass_hash_actual_raw[i]);
+    strcat(pass_hash_actual, temp);
+  }
+  return strcmp(pass_hash_valid, pass_hash_actual) == 0;
+}
+
+/**
+ * \brief check if the upload can be deleted, that is the user have
+ * the permission to delete this upload
+ *
+ * \param long upload_id - upload id
+ * \param char *user_name - user name
+ *
+ * \return 1: yes, you have the needed permissions;
+ *         0: no;
+ *        -1: failure;
+ *        -2: does not exist
+ */
+int check_permission_upload(int wanted_permissions, long upload_id, int user_id, int user_perm)
+{
+  int perms = GetUploadPermP(db_conn, upload_id, user_id, user_perm);
+  if (perms > 0)
+  {
+    if (perms < wanted_permissions)
+    {
+      return 0;
+    }
+    else
+    {
+      return 1;
+    }
+  }
+  return perms;
+}
+
+int check_read_permission_upload(long upload_id, int user_id, int user_perm)
+{
+  return check_permission_upload(PERM_READ, upload_id, user_id, user_perm);
+}
+
+int check_write_permission_upload(long upload_id, int user_id, int user_perm)
+{
+  return check_permission_upload(PERM_WRITE, upload_id, user_id, user_perm);
+}
+
+/**
+ * \brief check if the upload can be deleted, that is the user have
+ * the permission to delete this upload
+ *
+ * \param long upload_id - upload id
+ * \param char *user_name - user name
+ *
+ * \return 1: yes, can be deleted;
+ *         0: can not be deleted;
+ *        -1: failure;
+ */
+int check_write_permission_folder(long folder_id, int user_id, int user_perm)
+{
+  char SQL[MAXSQL];
+  PGresult *result;
+  int count = 0;
+
+  if (user_perm < PERM_WRITE)
+  {
+    return 0; // can not be deleted
+  }
+
+  snprintf(SQL,MAXSQL,"SELECT count(*) FROM folder join users on (users.user_pk = folder.user_fk or users.user_perm = 10) where folder_pk = %ld and users.user_pk = %d;",folder_id,user_id);
+  result = PQexec(db_conn, SQL);
+  if (fo_checkPQresult(db_conn, result, SQL, __FILE__, __LINE__))
+  {
+    return -1;
+  }
+  count = atol(PQgetvalue(result,0,0));
+  if(count == 0)
+  {
+    return 0; // can not be deleted
+  }
+  return 1; // can be deleted
+}
+
+/**
+ * \brief check if the upload can be deleted, that is the user have
+ * the permissoin to delete this upload
+ *
+ * \param long upload_id - upload id
+ * \param char *user_name - user name
+ *
+ * \return 1: yes, can be deleted;
+ *         0: can not be deleted;
+ */
+int check_write_permission_license(long license_id, int user_perm)
+{
+  if (user_perm != PERM_ADMIN)
+  {
+    verbosePrintf("only admin is allowed to delete licenses\n");
```

Comment:
this is not a self explanatory name. That would imply that it adds more information than a normal printf